cloudsync/tests/test_cs.py

```diff
@@ -819,6 +819,19 @@ def test_cs_basic(cs):
     assert not cs.state.changeset_len
 
 
+def test_cs_stop(cs):
+    cs.start()
+
+    with patch.object(cs.sthread, "join") as sthread_join:
+        cs.stop(wait=False)
+        assert cs.sthread
+        sthread_join.assert_not_called()
+
+        cs.stop(wait=True)
+        assert not cs.sthread
+        sthread_join.assert_called_once()
```

Comment:
It's important to ensure the thread is actually joined before the test completes

Reply:
the `stop_all()` call below ensures all 4 syncs are stopped and joined